From 8aa4bf9e8049ab1f0f4beda2eaa620f970f98692 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 12 Sep 2013 09:39:31 +0200
Subject: [PATCH 03/25] xhci: reset port when disabling slot

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1378978772-21612-4-git-send-email-kraxel@redhat.com>
Patchwork-id: 54336
O-Subject: [RHEL-7 qemu-kvm PATCH 3/4] xhci: reset port when disabling slot
Bugzilla: 1001604
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

Cc: qemu-stable@nongnu.org
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
(cherry picked from commit 5c67dd7b4884979a2613a4702ac1ab68b0e6a16e)
---
 hw/usb/hcd-xhci.c | 1 +
 1 file changed, 1 insertion(+)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/usb/hcd-xhci.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/hw/usb/hcd-xhci.c b/hw/usb/hcd-xhci.c
index ccf7cc5..0061cbe 100644
--- a/hw/usb/hcd-xhci.c
+++ b/hw/usb/hcd-xhci.c
@@ -2090,6 +2090,7 @@ static TRBCCode xhci_disable_slot(XHCIState *xhci, unsigned int slotid)
 
     xhci->slots[slotid-1].enabled = 0;
     xhci->slots[slotid-1].addressed = 0;
+    xhci->slots[slotid-1].uport = NULL;
     return CC_SUCCESS;
 }
 
-- 
1.7.1

